dev-packages/application-manager/src/generator/webpack-generator.ts

```diff
@@ -22,15 +22,15 @@ export class WebpackGenerator extends AbstractGenerator {
         return this.pck.resolveModulePath(moduleName, path).split(paths.sep).join('/');
     }
 
-    protected compileWebpackConfig(): string {
+    protected compileWebpackConfig(mode: string | undefined): string {
         return `// @ts-check
 const path = require('path');
```

Comment:
The minimal change is to generate into the webpack config:
```ts
const { mode }  = yargs.option('mode', {
    description: "Mode to use",
    choices: ["development", "production"],
    default: "production"
});
```
and pass `mode` into the configuration.

I don't see why other code should be changed and usage of `theia build` should stay simple by default.

Reply:
This is future-proof: we'll no longer be generating config file (and spawning a separate process) after #1552.

Reply:
We will discuss it later whether we want to implement https://github.com/theia-ide/theia/issues/1552 and how. For now, it is not necessary.